### ISSUE TITLE
Help Wanted: Update format of configuration settings

### ISSUE THREAD

**Issue by cwarnermm** [mattermost/docs#5951]
**Problem space**
~ 50% of Mattermost configuration settings are now published in a new format. 

The following pages of configuration settings need to be updated to the new format:

- https://docs.mattermost.com/configure/site-configuration-settings.html (WIP)
- https://docs.mattermost.com/configure/authentication-configuration-settings.html
- https://docs.mattermost.com/configure/plugins-configuration-settings.html
- https://docs.mattermost.com/configure/integrations-configuration-settings.html
- https://docs.mattermost.com/configure/compliance-configuration-settings.html
- https://docs.mattermost.com/configure/experimental-configuration-settings.html

**Target audiences**: System administrators, Support teams, integrators, developers

**Goal**: All Mattermost configuration settings are documented consistently using the same approach and format.

**Delivery expectations**
- 1 PR for each section of configuration updates; multiple contributors can work on this ticket at the same time.
- Use the [Environment configuration settings documentation](https://docs.mattermost.com/configure/environment-configuration-settings.html) and source code as a guideline when converting settings to the new table-based format.
- Plan & deployment badges and legacy edition support details will be incorporated by the Mattermost Docs team
- Comment in the PR to request a generated preview of your work. The Docs team will add the appropriate labels to your PR.

**Tech needed**
- System Console access; note that config settings can differ between a self-hosted and Cloud-based deployment
- a copy of the ``config.json`` file is helpful for looking up the equivalent configuration setting, but this is something that the Docs team can assist with
- RST format
- Sphinx

**Affected repository**: mattermost/docs

**Where to get help**
Join the Mattermost Docs team on the Mattermost Community Server in the [~Documentation Working Group channel](https://community.mattermost.com/core/channels/dwg-documentation-working-group) to discuss this ticket, or feel free to comment in this docs ticket.

**Comment by cwarnermm** [comment]
Thank you, @matthew-src! Really appreciate your help with this important work!

You've explained the work well. The tables are reStructuredText formatted. The description of the config setting is written in plain English for all audiences. 

Some existing configuration settings in the docs may be missing enough detail, such as a decent description, default values, possible values, and/or the full path to the config value in the ``config.json`` file (which also forms the environment variable). I can help fill in these missing details.

Really looking forward to your PR, @matthew-src.

**Comment by matthew-src** [comment]
I have made some progress, but I have a question regarding the notation for the ``config.json`` path. It seems that for ease of copy and pasting, the path should be given in dot notation without including the default value or other possible values. For Site Name this would be ``.TeamSettings.SiteName``, as opposed to `` ".TeamSettings.SiteName": "Mattermost"``.  What do you think?

(I did not see a consistent approach on the Environment Configuration page.)

**Comment by cwarnermm** [comment]
Great question, @matthew-src! And you're correct: the goal is to make copying and pasting easy for the user. 

The default values are included in dot notation in straightforward cases such as numeric and boolean values, and avoided for strings. This was done intentionally to prevent users from copying and pasting incorrect values; however, this approach has introduced an inconsistency in the dot notation content overall.

**Comment by cwarnermm** [comment]
I use a [tables generator](https://www.tablesgenerator.com/text_tables) site and enable the **Use reStructuredText syntax** option to make reasonably short work of RST table generation.

And I have a few tips to offer:
- When populating the table, include all RST formatting code too, such as ``**bolding**`` or link syntax. 
- Disable word wrap in your editor.
- Once you're finalizing the table content in an editor, working from left to right typically results in less table editing overall.
- If the table is missing from a preview, it can be a single table format character mis-aligned causing that to happen. Confirm that all parts of the table format code is out of place due to changes to spaces/text/code in the cells themselves.

**Comment by matthew-src** [comment]
Thank you. Would it be alright to open a work in progress PR to receive feedback on what I have done so far?

**Comment by matthew-src** [comment]
Hi @cwarnermm! I am **planning** to continue work on this issue in the next day or two, but I have some other tasks to complete before I start.

Below is a checklist of the remaining sections, which can be used to track the work being done by myself or others. (In the future, I will divide PRs by sections, as originally intended, instead of by page.) 

- [ ] Authentication
  - [ ] Signup
  - [ ] Email
  - [ ] Password
  - [ ] MFA
  - [ ] AD/LDAP
  - [ ] SAML 2.0
  - [ ] OAuth 2.0
  - [ ] OpenID Connect
  - [ ] Guest Access
- [ ] Plugins
  - [ ] Plugin Management
  - [ ] Agenda
  - [ ] Antivirus
  - [ ] Apps
  - [ ] Autolink
  - [ ] AWS SNS
  - [ ] Calls (beta)
  - [ ] Channel Export
  - [ ] Demo Plugin
  - [ ] GIF commands
  - [ ] Mattermost Boards
  - [ ] Mattermost Playbooks
  - [ ] User Satisfaction surveys
  - [ ] Zoom
- [ ] Integrations
  - [ ] Integrations management
  - [ ] Bot Accounts
  - [ ] GIF (beta)
  - [ ] CORS
- [ ] Compliance
  - [ ] Data Retention Policies
  - [ ] Compliance Export
  - [ ] Compliance Monitoring
  - [ ] Custom Terms of Service
- [ ] Experimental
  - [ ] Experimental System Console configuration settings
  - [ ] Experimental Bleve configuration settings
  - [ ] Experimental configuration settings for self-hosted deployments only
  - [ ] Experimental job configuration settings

**Comment by cwarnermm** [comment]
Really great to hear that, @matthew-src! Let me know if I can be of assistance!

**Comment by matthew-src** [comment]
I am taking a break from this PR. Anyone else is welcome to work on it. These lists of `config.json` keys and environment variables may be helpful:

[Mattermost config.json Keys](https://gist.github.com/matthew-src/7acb8ccb9d09f5b287fc9878f41fb601)

[Mattermost Configuration Environment Variables](https://gist.github.com/matthew-src/0d7d6eb9f7370658952182eaab10a2c7)

I also found this Vim plugin to be extremely helpful in working with tables: https://github.com/dhruvasagar/vim-table-mode.

**Comment by cwarnermm** [comment]
Closing this docs issue upon its completion.
